db/users.js

```diff
@@ -63,14 +64,14 @@ async function getUserById(id){
   }
 }
 async function getUserByUsername(userName){
-    try{
-        const {rows: [user] }= await client.query(`
-        SELECT * 
-        FROM users
-        WHERE username = $1; 
+  try{
+    const {rows: [user] }= await client.query(`
+      SELECT * 
+      FROM users
+      WHERE username = $1; 
     `, [userName]);
-   
-    return user;
+    delete user.password
```

Comment:
We should move this line into `getUser`, since we need the password to do the `bcrypt.compare`. It looks like we also call `getUserByUsername` in the `login` and `register` routes, so we should also add this line there. 

Alternatively, you can add another parameter that is not required and has a default value of false, `async function getUserByUsername(userName, includePassword=false){`
Then add a conditional:
`if (!includePassword) delete user.password;`

Then you only need to update `getUser` to call `getUserByUsername(username, true)` and then after the `bcrypt.compare` to call `delete user.password`

Does that make sense?